index.js

```diff
@@ -237,6 +237,37 @@ const execa = (command, args, options) => {
 		return Promise.reject(error);
 	}
 
+	const originalKill = spawned.kill.bind(spawned);
+	spawned.kill = (signal, options) => {
+		if (signal === undefined ||
+			(typeof signal === 'string' && signal.toUpperCase() === 'SIGTERM') ||
+			(typeof signal === 'number' && signal === os.constants.signals.SIGTERM)) {
+			const retryAfter = options !== undefined &&
+				['number', 'string'].includes(typeof options.retryAfter) ?
+				Number(options.retryAfter) :
+				5000;
+
+			const retry = !(options !== undefined && options.retry === false);
+
+			const killResult = originalKill(signal);
+
+			if (retry && killResult) {
+				const safetyTimeout = setTimeout(() => {
```

Comment:
This is a tiny optimization, but `retryAfter` can now be declared within this block, so it's not evaluated if this block is not evaluated.